exchange/utils_test.go

```diff
@@ -2609,6 +2609,27 @@ func TestBuildXPrebidHeader(t *testing.T) {
 	}
 }
 
+func TestSourceExtSChainCopied(t *testing.T) {
+	bidRequest := newBidRequest(t)
+
+	bidderSchains := map[string]*openrtb_ext.ExtRequestPrebidSChainSChain{
+		"bidder1": {
+			Complete: 1,
+		},
+		"bidder2": {
+			Complete: 2,
+		},
+	}
+
+	copy1 := *bidRequest
+	prepareSource(&copy1, "bidder1", bidderSchains)
+	copy2 := *bidRequest
+	prepareSource(&copy2, "bidder2", bidderSchains)
+
+	assert.Equal(t, json.RawMessage(`{"schain":{"complete":1,"nodes":null,"ver":""}}`), copy1.Source.Ext, "First schain was overwritten or not set")
+	assert.Equal(t, json.RawMessage(`{"schain":{"complete":2,"nodes":null,"ver":""}}`), copy2.Source.Ext, "Second schain was overwritten or not set")
```

Comment:
Thank you @bjorn-lw . One last ask: could we also assert that the primitive values are preserved when copying the Source field?
```
2547   func TestSourceExtSChainCopied(t *testing.T) {
2548       bidRequest := newBidRequest(t)
2549  
2550       bidderSchains := map[string]*openrtb_ext.ExtRequestPrebidSChainSChain{
2551           "bidder1": {
2552               Complete: 1,
2553           },
2554           "bidder2": {
2555               Complete: 2,
2556           },
2557       }
2558  
2559       copy1 := *bidRequest
2560       prepareSource(&copy1, "bidder1", bidderSchains)
2561       copy2 := *bidRequest
     +     copy2.Source.TID = "Different value"
2562       prepareSource(&copy2, "bidder2", bidderSchains)
2563  
2564       assert.Equal(t, json.RawMessage(`{"schain":{"complete":1,"nodes":null,"ver":""}}`), copy1.Source.Ext, "First schain was overwritten or not set")
2565       assert.Equal(t, json.RawMessage(`{"schain":{"complete":2,"nodes":null,"ver":""}}`), copy2.Source.Ext, "Second schain was overwritten or not set")
     +
     +     // Assert copy1.Source was copied inside prepareSource() and primitive values were not overwritten
     +     assert.NotEqual(t, copy1.Source.TID, copy2.Source.TID, "Primitive values inside copy1.Source are not supposed get overwritten")
2566   }
exchange/utils_test.go
```

Reply:
@guscarreon Extended the test as you asked for.